### PR TITLE
Full text search productions

### PR DESCRIPTION
**Pull request by jdrieghe**
### Added
- Endpoint to search productions by keyword
- Fulltext index on `name` field of productions

### CODE REVIEW

File: src/Http/Productions/ProductionsSearchController.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+     * @var ProductionRepository
+     */
+    private $repository;
+
+    public function __construct(ProductionRepository $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    public function search(Request $request): Response
```
Comment (bertramakers) [inline src/Http/Productions/ProductionsSearchController.php:23]: I was thinking we should probably add pagination to this request to be safe?

If we do, I'd follow the `start` and `limit` parameters we already use in various places like [Search API3](https://github.com/cultuurnet/udb3-search-service/blob/master/src/Http/OfferSearchController.php#L143-L144) and the [labels query endpoint](https://github.com/cultuurnet/udb3-silex/blob/master/src/Http/Label/Query/QueryFactory.php#L14-L15) for consistency.

For the response data, you can use the existing [`CultuurNet\Hydra\PagedCollection` class](https://github.com/cultuurnet/Hydra/blob/master/src/PagedCollection.php). For an example of how to use it, you can look at [`CultuurNet\UDB3\Http\Label::createPagedCollection()`](https://github.com/cultuurnet/udb3-silex/blob/master/src/Http/Label/ReadRestController.php#L101-L125) (a private method and specific to that controller, but a decent example). That way we maintain a consistent response format for paginated requests based on a w3 standard.

If you have any more questions how to implement this, let me know!

**Comment by bertramakers** [comment]
The overall structure of the code looks good as always, and I see no major issues. Just think we should add the pagination to prevent huge result sets for common keywords.

**Comment by bertramakers** [comment]
Oh, also when all of these endpoints are done and merged we should document them.
There's two Swagger docs, one for public docs and one for internal docs. I think productions belongs in the internal docs, which are here: https://github.com/cultuurnet/udb3-swagger/ 
Sadly the naming of the repo's is a bit ambiguous. I'll add some documentation on Confluence where each categorie of docs lives for each API.

**Comment by jdrieghe** [comment]
@bertramakers as discussed on Slack, I passed on the `PagedCollection` for now and added a default and configurable limit to number of search results.

File: app/Migrations/Version20200618120205.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+use Doctrine\DBAL\Schema\Schema;
+
+class Version20200618120205 extends AbstractMigration
+{
+    public function up(Schema $schema): void
+    {
+        $table =  $schema->getTable('productions');
+        // This is the previously added index on 'name' which we no longer need
+        $table->dropIndex('idx_bbd7c0c25e237e06');
+        $table->addIndex(['name'], 'idx_search_name', ['fulltext']);
```
[the remark this answers is not in the document]
Comment (jdrieghe) [reply app/Migrations/Version20200618120205.php:15]: Yep, that's exactly what I expected and what I got in my testing as well. 👍 
Thanks for the double check!

**Pull request merged**